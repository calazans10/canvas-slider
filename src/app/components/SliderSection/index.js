import PropTypes from 'prop-types';
import Slider from '../Slider';
import Section from './style';

function SliderSection({ width, height, images }) {
  return (
    <Section>
      <Slider width={width} height={height} images={images} />
      <Section.Aside>
        <Section.Text>Drag to change image</Section.Text>
      </Section.Aside>
    </Section>
  );
}

SliderSection.propTypes = {
  width: PropTypes.number.isRequired,
  height: PropTypes.number.isRequired,
  images: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default SliderSection;
