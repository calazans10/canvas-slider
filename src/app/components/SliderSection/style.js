import styled from 'styled-components';

const Section = styled.section`
  text-align: center;
`;

Section.Aside = styled.aside`
  margin-top: 8px;
`;

Section.Text = styled.p`
  color: #999;
  font-size: 12px;
`;

export default Section;
