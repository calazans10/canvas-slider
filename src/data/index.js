import bob0 from '../images/0.jpg';
import bob1 from '../images/1.jpg';
import bob2 from '../images/2.jpg';
import bob3 from '../images/3.jpg';
import bob4 from '../images/4.jpg';
import bob5 from '../images/5.jpg';
import bob6 from '../images/6.jpg';

// eslint-disable-next-line import/prefer-default-export
export const images = [bob0, bob1, bob2, bob3, bob4, bob5, bob6];
