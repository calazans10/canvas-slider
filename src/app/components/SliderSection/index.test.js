import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { images } from '../../../data';
import SliderSection from '.';

describe('SliderSection', () => {
  test('renders without crashing', async () => {
    const wrapper = shallow(<SliderSection width={640} height={400} images={images} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
