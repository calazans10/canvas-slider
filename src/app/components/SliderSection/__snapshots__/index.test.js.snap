// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderSection renders without crashing 1`] = `
<styled.section>
  <Slider
    height={400}
    images={
      Array [
        "0.jpg",
        "1.jpg",
        "2.jpg",
        "3.jpg",
        "4.jpg",
        "5.jpg",
        "6.jpg",
      ]
    }
    width={640}
  />
  <styled.aside>
    <styled.p>
      Drag to change image
    </styled.p>
  </styled.aside>
</styled.section>
`;
